File: puzzles/python3/horse-racing-duals/horse_racing_duals.py
from typing import List


if __name__ == "__main__":
    strengths: List[int] = []
    min_diff = 10000000
    n = int(input())
    for _ in range(n):
        strengths.append(int(input()))

    strengths.sort()

    for i in range(n - 1):
        diff = strengths[i + 1] - strengths[i]
        if diff < min_diff:
            min_diff = diff

    print(min_diff)

File: puzzles/swift/horse-racing-duals/horse-racing-duals.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
    public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

if let n = readLine(), let nInt = Int(n) {
    var strengths: [Int] = []
    var minDiff = 10000000

    for _ in 0..<nInt {
        if let strengthStr = readLine(), let strength = Int(strengthStr) {
            strengths.append(strength)
        }
    }

    strengths.sort()

    for i in 0..<(nInt - 1) {
        let diff = strengths[i + 1] - strengths[i]
        if diff < minDiff {
            minDiff = diff
        }
    }

    print(minDiff)
}
